_posts/2022-06-23-cockpit-272.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Cockpit 272
+author: garrett
+date: '2022-06-23'
+tags: cockpit
+slug: cockpit-272
+category: release
+summary: 'Edit custom firewall ports, pin favorite services, dark login support, and unprivilaged cockpit/ws mode'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 272:
+
+## Firewall: Edit custom services
+
+The Firewall page has been able to open custom ports by adding creating custom services. This release adds the ability to edit custom services.
+
+![edit-custom]({{ site.baseurl }}/images/272-firewall-edit-custom-services.png)
+
+## Services: Pin services as favorites
+
+The services page now allows users to pin any service to the top of the services list.
+
+![pin-2]({{ site.baseurl }}/images/272-services-pin-services-as-favorites.png)
+
+To pin a service, navigate to its detail page and click `Pin unit` in the menu next to its name.
+
+![pin]({{ site.baseurl }}/images/272-services-pin-services-as-favorites-2.png)
+
+## Login: Dark mode
+
+The login page now has a dark mode which changes with your system's dark settings. Most desktops have a setting for this in "appearance" area in system settings, including GNOME, KDE, Windows, macOS, Android, and iOS/iPadOS.
+
+Here's [Cockpit Client](https://flathub.org/apps/details/org.cockpit_project.CockpitClient), which [also recently got dark mode a few releases ago](https://cockpit-project.org/blog/cockpit-269.html), in the standard light mode:
+
+![Screenshot from 2022-06-22 18-42-47]({{ site.baseurl }}/images/272-login-dark-mode.png)
+
+And after switching to dark settings, it looks like this:
+
+![Screenshot from 2022-06-22 18-42-59]({{ site.baseurl }}/images/272-login-dark-mode-2.png)
+
+We’re still working on adding dark mode for Cockpit once you've logged in, in the near future.
+
+## Unprivileged cockpit/ws container mode
+
+The [cockpit/ws container](https://quay.io/repository/cockpit/ws) can now run in unprivileged mode.
+
+It presents an unbranded variant of login page that always asks for a host name. Connections are made with SSH.
+
+This mode is suitable for deploying to, e.g., Kubernetes or similar environments, where you cannot have or want privileged containers. In this "bastion host mode", you can have Cockpit for servers in your data center without opening an extra port for `cockpit-ws`.
+
+Currently, username + password and "classic RSA" type SSH keys are supported. See the container documentation for details.
```

Comment:
Would prefer to see "username and password" written out, but it's not a big deal.

Reply:
Yeah, the problem is the string of ands. It becomes ambiguous. It was previously username/password, but it's not username _or_ password, so that doesn't really work.